freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/VaultPathBuilder.java

```diff
@@ -44,14 +45,16 @@ public VaultPathBuilder withSubType(String subType) {
     }
 
     public VaultPathBuilder withEnvironmentCrn(String environmentCrn) {
-        String envCrn = StringUtils.substringAfterLast(environmentCrn, ":");
-        this.environmentCrn = Optional.ofNullable(envCrn);
+        this.environmentId = Optional.ofNullable(environmentCrn).map((String crn) -> {
+            return Crn.safeFromString(crn).getResource();
+        });
         return this;
     }
 
     public VaultPathBuilder withClusterCrn(String clusterCrn) {
-        String crn = StringUtils.substringAfterLast(clusterCrn, ":");
-        this.clusterCrn = Optional.ofNullable(crn);
+        this.clusterId = Optional.ofNullable(clusterCrn).map((String crn) -> {
+            return Crn.safeFromString(crn).getResource();
```

Comment:
while `StringUtils.substringAfterLast` won't throw exception, `Crn.safeFromString` can throw. Shouldn't we handle this?